server/routes.ts

```diff
@@ -31,14 +30,13 @@ router.get("/system/health", (req, res) => {
   });
 });
 
-// Admin
+router.get("/quest", checkInteractiveCredentials, handleGetQuestDetails);
```

Comment:
we are checkInteractiveCredentials for every request, that could take up to 0.4 to 1 sec more, per request, making the app slow for the end user

Reply:
I do think we need this check this and the middleware simply checks for matching keys so is really lightweight and fast but you just inspired me to move it to the getCredentials util which will clean this up a little. Thanks!